File: server/src/main/java/de/ingrid/igeserver/documenttypes/AddressWrapperType.java
package de.ingrid.igeserver.documenttypes;

import com.orientechnologies.orient.core.db.ODatabaseSession;
import com.orientechnologies.orient.core.metadata.schema.OClass;
import com.orientechnologies.orient.core.metadata.schema.OSchema;
import com.orientechnologies.orient.core.metadata.schema.OType;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.stereotype.Service;

@Service
public class AddressWrapperType extends DocumentType {

    private static Logger log = LogManager.getLogger(AddressWrapperType.class);

    public static final String TYPE = "AddressWrapper";

    private static final String[] profiles = new String[0];

    public AddressWrapperType() {
        super(TYPE, profiles);
    }

    @Override
    public void initialize(ODatabaseSession session) {

        OSchema schema = session.getMetadata().getSchema();
        if (!schema.existsClass(TYPE)) {
            log.debug("Create class " + TYPE);
            OClass docClass = schema.createClass(TYPE);

            // TODO: set more constraints and information for a new catalog (name, email?, ...)
            docClass.createProperty("_id", OType.STRING);
            docClass.createProperty("_parent", OType.STRING);
            docClass.createProperty("draft", OType.LINK);
            docClass.createProperty("published", OType.LINK);
            docClass.createProperty("archive", OType.LINKLIST);
        }

    }

}

File: server/src/main/java/de/ingrid/igeserver/documenttypes/AddressWrapperType.kt
package de.ingrid.igeserver.documenttypes

import com.orientechnologies.orient.core.db.ODatabaseSession
import com.orientechnologies.orient.core.metadata.schema.OType
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.kotlin.logger
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Service

@Service
@Order(1)
class AddressWrapperType : DocumentType(TYPE, profiles) {

    val log = logger()

    companion object {
        const val TYPE = "AddressWrapper"
        private val profiles = arrayOf<String>()
    }

    override fun initialize(session: ODatabaseSession) {
        val schema = session.metadata.schema
        if (!schema.existsClass(TYPE)) {
            log.debug("Create class $TYPE")
            val docClass = schema.createClass(TYPE)

            // TODO: set more constraints and information for a new catalog (name, email?, ...)
            docClass.createProperty("_id", OType.STRING)
            docClass.createProperty("_parent", OType.STRING)
            docClass.createProperty("draft", OType.LINK)
            docClass.createProperty("published", OType.LINK)
            docClass.createProperty("archive", OType.LINKLIST)
        }
    }

}